cs411project/sql/check-machine-availability.sql

```diff
@@ -0,0 +1,83 @@
+/*
+ To test these queries, source heartbeat-insert-test.sql within the MySQL prompt
+  and then source this file. You should compare the output of this file versus
+  the expected results denoted in heartbeat-insert-test.sql. Note that the
+  test uses CURRENT_TIMESTAMP and these queries also user CURRENT_TIMESTAMP,
+  so it is important to run this query after inserting the test records because
+  if you wait too long (4 min), then all machines will be returned as 
+  AVAILABLE (since all heartbeats have timed out w.r.t. CURRENT_TIMESTAMP
+  of a later time) 
+*/
+
+
+
+/* Check the availability of a single machine (7 for this example) */
+/* Assuming for Machine.Status that 0 is BROKEN, 1 is ALIVE */
+SELECT
```

Comment:
For both the bulk and single-machine queries to check for availability, if there is no HeartbeatSequence row for a machine M, then the query returns BROKEN if the machine's status is BROKEN, otherwise the query returns AVAILABLE (which is what we want because if there is no HeartbeatSequence row for M, then no one has used the machine before). The three-valued Boolean logic works in our favor here